jenkins/opensearch-dashboards/Jenkinsfile

```diff
@@ -131,6 +131,19 @@ pipeline {
                 success {
                     node('Jenkins-Agent-al2-x64-c54xlarge-Docker-Host') {
                         publishNotification(":white_check_mark:", "Successful Build", "\n${getAllJenkinsMessages()}")
```

Comment:
Same as for #928, this notifies before it finishes, potentially hiding the error/failing the workflow in a success step.

Any reason why building a docker image isn't a regular step in the workflow?

Reply:
> Same as for #928, this notifies before it finishes, potentially hiding the error/failing the workflow in a success step.
> 
> Any reason why building a docker image isn't a regular step in the workflow?

Because I think we don't wish to build the docker image if the bundle build workflow fails. Furthermore, our approach to trigger the `docker-build` workflow will take the tarballs from successful bundle build.

Reply:
In the pipeline workflow there are serial and parallel stages.  If in series, if a previous stage fails it will prevent the docker image from being created.  Creating docker image does not have to happen via job invocation as done here, it can be included in this pipeline as a stage, which this is a form of code sharing the code @dblock just created a code review for a much more testable mechanism.

We have written in these jenkinsfiles in a fragile way for haste, great bias for action.  Lets update this code to be more maintainable by creating an issue to clean up the following areas: 
* Use the generated manifest files to lookup the artifacts, removing duplication of url generation, stashing the manifest would be appropriate
* Move docker image generation into a new stage and add notifications after it has finished
* Move docker image generation code into shared library

I recommend we do these one at a time to ease troubleshooting and speed it up.